sdk/spring/azure-spring-boot-test-aad/src/test/java/com/azure/test/aad/selenium/AADLoginRunner.java

```diff
@@ -0,0 +1,207 @@
+
+public class AADLoginRunner {
+
+    public static final String DEFAULT_USERNAME = System.getenv(AAD_USER_NAME_1);
+    private static final String DEFAULT_PASSWORD = System.getenv(AAD_USER_PASSWORD_1);
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AADLoginRunner.class);
+
+    static {
+        final String directory = "src/test/resources/driver/";
+        final String chromedriverLinux = "chromedriver_linux64";
+        final String chromedriverWin32 = "chromedriver_win32.exe";
+        final String chromedriverMac = "chromedriver_mac64";
+        String osName = System.getProperty("os.name").toLowerCase();
+        Process process = null;
+        try {
+            File dir = new File(directory);
+            if (Pattern.matches("linux.*", osName)) {
+                process = Runtime.getRuntime().exec("chmod +x " + chromedriverLinux, null, dir);
+                process.waitFor();
+                System.setProperty(ChromeDriverService.CHROME_DRIVER_EXE_PROPERTY, directory + chromedriverLinux);
+            } else if (Pattern.matches("windows.*", osName)) {
+                System.setProperty(ChromeDriverService.CHROME_DRIVER_EXE_PROPERTY, directory + chromedriverWin32);
+            } else if (Pattern.matches("mac.*", osName)) {
+                process = Runtime.getRuntime().exec("chmod +x " + chromedriverMac, null, dir);
+                process.waitFor();
+                System.setProperty(ChromeDriverService.CHROME_DRIVER_EXE_PROPERTY, directory + chromedriverMac);
+            } else {
+                throw new IllegalStateException("Can not recognize osName. osName = " + System.getProperty("os"
+                    + ".name"));
+            }
+        } catch (InterruptedException | IOException e) {
+            throw new RuntimeException(e);
+        } finally {
+            if (process != null) {
+                process.destroy();
+            }
+        }
+    }
+
+    private final AppRunner app;
+    private final WebDriver driver;
+    private final String password;
+    private final String username;
+    private final WebDriverWait wait;
+
+    private AADLoginRunner(String username, String password, AppRunner app, WebDriver driver) {
+        this.username = username;
+        this.password = password;
+        this.app = app;
+        this.driver = driver;
+        this.wait = new WebDriverWait(this.driver, 10);
+    }
+
+    public static AADLoginRunnerConfiguration build(Class<?> appClass) {
+        return new AADLoginRunnerConfiguration(appClass);
+    }
+
+    public void run(BrowserCommandWithAppRunner command) {
+        try {
+            this.app.start();
+            command.login(login())
+                   .andThen((app, driver, wait) -> LOGGER.info("Test ===> {}.{}() has finished running.",
+                       Thread.currentThread().getStackTrace()[6].getClassName(),
+                       Thread.currentThread().getStackTrace()[6].getMethodName()))
```

Comment:
`6` is not good, if we and more methods in the stack trace, it may change to `7` or `8`.